src/caret_analyze/architecture/architecture.py

```diff
@@ -711,18 +722,39 @@ def update_message_context(self, context_type: str,
             if (context['subscription_topic_name'], context['publisher_topic_name']) ==\
                (subscribe_topic_name, publish_topic_name):
                 context['context_type'] = context_type
-                break
-        else:
-            self._contexts.append({'context_type': context_type,
-                                   'subscription_topic_name': subscribe_topic_name,
-                                   'publisher_topic_name': publish_topic_name})
-
-    def remove_callback_chain(self, subscribe_topic_name: str, publish_topic_name: str) -> None:
+        for path in self._node.paths:
+            if path.message_context is not None:
+                # already processed by the above self._contexts process
+                continue
+            else:
+                if (path.subscribe_topic_name, path.publish_topic_name) ==\
+                        (subscribe_topic_name, publish_topic_name):
+                    self._contexts.append({'context_type': context_type,
+                                           'subscription_topic_name': subscribe_topic_name,
+                                           'publisher_topic_name': publish_topic_name,
+                                           'publisher_construction_order':
+                                           path.publisher_construction_order,
+                                           'subscription_construction_order':
+                                           path.subscription_construction_order})
```

Comment:
Suggestion to improve readability.
```suggestion
                    self._contexts.append({
                        'context_type': context_type,
                        'subscription_topic_name': subscribe_topic_name,
                        'publisher_topic_name': publish_topic_name,
                        'publisher_construction_order': path.publisher_construction_order,
                        'subscription_construction_order': path.subscription_construction_order
                    })
```